src/arc_utilities/ros_helpers.py

```diff
@@ -29,10 +29,67 @@ def callback(self, msg):
         with self.lock:
             self.data = msg
 
-    def get(self):
+    def get(self, block_on_none=False):
```

Comment:
I think `block_on_none=False` is fine here; if we want a different default I'd put it in the constructor rather than `get()`; typical use would be to only `block_on_none` once, and object construction is the time that makes the most sense to me.

I can think of one use case where we might want to not block on construction and that would be the case where we are creating multiple objects that need to wait for valid data before being used, at which point it makes sense to create them all, then do a wait block for all of them, but I think this case would be the execption rather than the normal use, so making the constructor block by default would be my suggestion.

Reply:
We do NOT want to block in the constructor. We don't care if we get data until we actually want to use it. 
If we block in the constructor we add initialization time for each Listener we create.
If we block in the get method, any time between construction and "get" is used to wait for data. We perhaps have already even gotten data by then, so no time is wasted.

The scenario you mention will occur frequently (I expect). We will want to create multiple listeners and there is no point in creating one, waiting, creating a second, waiting... To create them all then be required to run a "Wait for data" for each of them would be syntactically more confusing, more cumbersome, and more error prone.

Reply:
We could however put the argument `block_on_none` in the constructor, but still do the check in the `get` function.

Reply:
Right, that's what I had in mind. I think the defaults should be such that
```Python
listener = Listener("topic", MessageType)
data = listener.get()
```
will return a valid object. Two ways of doing this. `get()` defaults to blocking behavior, or `Listener(...)` defaults to blocking behavior (or both do). I prefer having `Listener(...)` default to blocking with a flag to override that makes more sense, and we can always get the other behavior as well by passing `block_on_none=True` to `get()`.

Reply:
Ya, I'm liking it better in the constructor as well. I'll change.